### ISSUE TITLE
 Crash when notebook computes T*VV with T a circulant matrix and VV = V + V a vector 

### ISSUE THREAD

**Issue by fp4code** [fonsp/Pluto.jl#2211]
Pluto cell ```T*VV``` crashes, where ```T``` is a ToeplitzMatrices.Circulant and ```VV = V + V``` a vector. Strangely ```T*V``` is OK.

Same behavior with julia 1.6, 1.7, 1.8, master

No problem when directly executing the notebook as ```julia Pluto-Circulant-Bug.jl```. Crashes only as a Pluto notebook.

**The notebook:** https://gist.github.com/fp4code/0028cd0458044421470d9a7f4301178f

**The crash message:**

```
MethodError: convert(::Type{AbstractArray{Float64}}, ::ToeplitzMatrices.Circulant{Float64}) is ambiguous.

Candidates:
convert(::Type{T}, a::T) where T<:AbstractArray
@ Base abstractarray.jl:16
convert(::Type{AbstractArray{T}}, S::ToeplitzMatrices.AbstractToeplitz) where T
@ ToeplitzMatrices ~/.julia/packages/ToeplitzMatrices/bWAGV/src/ToeplitzMatrices.jl:43
Possible fix, define
convert(::Type{AbstractArray{T}}, ::ToeplitzMatrices.AbstractToeplitz{T}) where T<:Number
1. setindex!(::Vector{AbstractArray{Float64}}, ::ToeplitzMatrices.Circulant{Float64}, ::Int64)@array.jl:959
2. restart_copyto_nonleaf!(::Vector{AbstractArray{Float64}}, ::Vector{ToeplitzMatrices.Circulant{Float64}},
::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{1}, Tuple{Base.OneTo{Int64}}, typeof(getfield), Tuple{Base.Broadcast.Extruded{Vector{Module}, Tuple{Bool}, Tuple{Int64}}, Base.Broadcast.Extruded{Vector{Symbol}, Tuple{Bool}, Tuple{Int64}}}}, ::Vector{Float64}, ::Int64, ::Base.OneTo{Int64}, ::Int64, ::Int64)@broadcast.jl:1056
3. copyto_nonleaf!(::Vector{ToeplitzMatrices.Circulant{Float64}}, ::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{1}, Tuple{Base.OneTo{Int64}}, typeof(getfield), Tuple{Base.Broadcast.Extruded{Vector{Module}, Tuple{Bool}, Tuple{Int64}}, Base.Broadcast.Extruded{Vector{Symbol}, Tuple{Bool}, Tuple{Int64}}}}, ::Base.OneTo{Int64}, ::Int64, ::Int64)@broadcast.jl:1075
4. copy@broadcast.jl:920[inlined]
5. materialize@broadcast.jl:873[inlined]
6. compute(::Module, ::Main.PlutoRunner.Computer)@PlutoRunner.jl:402
7. (::Main.PlutoRunner.var"#31#35"{Module})()@PlutoRunner.jl:574
8. run_inside_trycatch(::Module, ::Main.PlutoRunner.var"#31#35"{Module})@PlutoRunner.jl:445
9. (::Main.PlutoRunner.var"#29#33"{Bool, Module, Expr, Base.UUID, Base.UUID, Tuple{Set{Symbol}, Set{Symbol}}, Nothing, UInt64, Expr})()@PlutoRunner.jl:574
10. var"#with_io_to_logs#93"(::Bool, ::Base.CoreLogging.LogLevel, ::typeof(Main.PlutoRunner.with_io_to_logs), ::Main.PlutoRunner.var"#29#33"{Bool, Module, Expr, Base.UUID, Base.UUID, Tuple{Set{Symbol}, Set{Symbol}}, Nothing, UInt64, Expr})@PlutoRunner.jl:2192
11. (::Main.PlutoRunner.var"#28#32"{Bool, Bool, Module, Expr, Base.UUID, Base.UUID, Tuple{Set{Symbol}, Set{Symbol}}, Nothing, UInt64, Expr})()@PlutoRunner.jl:549
12. with_logstate(::Function, ::Any)@logging.jl:514
13. with_logger@logging.jl:626[inlined]
14. var"#run_expression#27"(::Bool, ::Bool, ::typeof(Main.PlutoRunner.run_expression), ::Module, ::Expr, ::Base.UUID, ::Base.UUID, ::Union{Nothing, Tuple{Set{Symbol}, Set{Symbol}}}, ::Union{Nothing, UInt64})@PlutoRunner.jl:548
15. top-level scope@none:1
```

**Comment by fp4code** [comment]
A simpler way to crash a cell is to define
```
VV = Vector(V)
```
and then
```
T*VV
```
crashes.

Strangely
```
T*Vector(VV)
```
doesn't crash.

```
VV = Vector(V) ; T*VV
```
is also OK.

**Comment by Pangoraw** [comment]
Thanks for the detailed investigation, it should be fixed by #2212.

**Comment by fp4code** [comment]
Good fix! Stunned by your responsiveness @Pangoraw 😊
